File: InterviewBit/Array/MinStepInInfiniteGrid/minStepInInfiniteGrid.py
class Solution:
    # @param A : list of integers
    # @param B : list of integers
    # @return an integer
    
    def steps(self, x,y,x_next,y_next):
        if (x,y) == (x_next, y_next):
            return 0
        else:
            return max(abs(x-x_next),abs(y-y_next))
                
    def coverPoints(self, A, B):
        count = 0
        n = len(A)
        for i in range(n-1):
            x, y = A[i], B[i]
            x_next, y_next = A[i+1], B[i+1]
            count += self.steps(x,y,x_next, y_next)
        return count

File: InterviewBit/Array/MinStepInInfiniteGrid/minStepInInfiniteGrid.cpp
#define loop(a,b) for(int i=a;i<b;i++)

int steps(int cx, int cy, int tx, int ty){
    if(cx==tx&&cy==ty) return 0;
    int xd = abs(cx-tx);
    int yd = abs(cy-ty);
    return int(max(xd,yd));
}
int Solution::coverPoints(vector<int> &A, vector<int> &B) {

int n=0;
    int x=A[0];
    int y=B[0];
    int size = A.size();
    loop(1,size){
        int xt = A[i];
        int yt = B[i];
        n += steps(x,y,xt,yt);
        x = xt;
        y = yt;
    }
    return n;
    
}
